src/ngamsPlugIns/ngamsPlugIns/ngamsSdmMultipart.py

```diff
@@ -58,168 +62,151 @@ def specificTreatment(fo):
         try:
-            almaUid = message["Content-Location"]
-        except:
-            err = "Mandatory alma-uid or Content-Location parameter not found in mime header!"
-            errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                               _PLUGIN_ID, err])
-            raise Exception(errMsg)
-
-    if not uidTempl.match(almaUid):
-        err = "Invalid alma-uid found in Content-Location: " + almaUid
-        errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                               _PLUGIN_ID, err])
-        raise Exception(errMsg)
-
-# Now, build final filename. We do that by looking for the UID in
-# the message mime-header.
-
-# The final filename is built as follows: <almaUidR>.<ext>
-# where almaUidR has the slash character in the UID replaced by colons.
+            alma_uid = message["Content-Location"]
+        except Exception:
+            error = "Mandatory alma-uid or Content-Location parameter not found in mime header!"
+            error_message = genLog("NGAMS_ER_DAPI_BAD_FILE",
+                            [os.path.basename(filename), _PLUGIN_ID, error])
+            raise Exception(error_message)
+
+    # This matches the old UID of type X0123456789abcdef/X01234567
+    # uid_expression = re.compile("[xX][0-9,a-f]{3,}/[xX][0-9,a-f]{1,}$")
+
+    # This matches the new UID structure uid://X1/X2/X3#kdgflf
+    # uid_expression = re.compile("^[uU][iI][dD]:/(/[xX][0-9,a-f,A-F]+){3}(#\w{1,}|/\w{0,}){0,}$")
+
+    # Update for the new assignment of archiveIds (backwards compatible)
+    uid_expression = re.compile("^[uU][iI][dD]://[aAbBcCzZxX][0-9,a-z,A-Z]+(/[xX][0-9,a-z,A-Z]+){2}(#\w{1,}|/\w{0,}){0,}$")
+
+    if not uid_expression.match(alma_uid):
+        error = "Invalid alma-uid found in Content-Location: " + alma_uid
+        error_message = genLog("NGAMS_ER_DAPI_BAD_FILE",
+                               [os.path.basename(filename), _PLUGIN_ID, error])
+        raise Exception(error_message)
+
+    # Now, build final filename. We do that by looking for the UID in
+    # the message mime-header.
+    #
+    # The final filename is built as follows: <almaUidR>.<ext>
+    # where almaUidR has the slash character in the UID replaced by colons.
     try:
-        # get rid of the 'uid://' and of anything following a '#' sign
-        almaUid = almaUid.split('//',2)[1].split('#')[0]
-        if almaUid[-1] == '/': almaUid = almaUid[:-1]  #remove trailing /
+        # Get rid of the 'uid://' and of anything following a '#' sign
+        alma_uid = alma_uid.split('//', 2)[1].split('#')[0]
 
-        fileId = almaUid
-        finalFileName = almaUid.replace('/',':')
-# Have no idea why, but the extension is added somewhere else...
-#        if os.path.splitext(finalFileName)[-1] != _EXT:
-#            finalFileName += _EXT
+        # Remove trailing '/'
+        if alma_uid[-1] == '/':
+            alma_uid = alma_uid[:-1]
```

Comment:
`alma_uid = alma_uid.rstrip("/")` maybe expresses this more compactly?

Reply:
Yes, this is much cleaner